src/lyapunov_spectrum.jl

```diff
@@ -73,62 +77,90 @@ function specular!(p::Particle{T}, o::Union{InfiniteWall{T},FiniteWall{T}}, offs
         offset[:,k] = x
     end
 end
+
+
 """
-    resolvecollision!(p::Particle, o::Union{Disk, InfiniteWall}, offset::MArray)
+    resolvecollision!(p::AbstractParticle, o::Union{Disk, InfiniteWall}, offset::MArray)
 Resolve the collision between particle `p` and obstacle `o` of type *Circular*,
 updating the components of the offset vectors stored in the matrix `offset` as columns.
 """
-function resolvecollision!(p::AbstractParticle{T}, o::Union{Disk{T}, InfiniteWall{T}, FiniteWall{T}, Semicircle{T}}, offset::MArray{Tuple{4,4}, T})::Void where {T<:AbstractFloat}
+function resolvecollision!(p::AbstractParticle{T},
+                o::Union{Disk{T}, InfiniteWall{T}, FiniteWall{T}, Semicircle{T}},
+                offset::MArray{Tuple{4,4}, T})::Void where {T<:AbstractFloat}
+
     specular!(p, o, offset)
     return nothing
 end
 
-resolvecollision!(p::AbstractParticle{T}, o::PeriodicWall{T}, offset::MArray{Tuple{4,4}, T}) where {T<:AbstractFloat} =
-resolvecollision!(p, o)
+resolvecollision!(p::AbstractParticle{T}, o::PeriodicWall{T},
+                  offset::MArray{Tuple{4,4}, T}) where {T} = resolvecollision!(p, o)
 
 """
-    propagate!(p::Particle{T}, t::T, offset::MArray{Tuple{4,4},T})
-Propagate the particle `p` for given time `t`, changing appropriately the the
-`p.pos` and `p.vel` fields together with the components of the offset vectors
-stored in the `offset` matrix.
+    propagate_offset!(offset::MArray{Tuple{4,4},T}, p::AbstractParticle)
+Computes the linearized evolution of the offset vectors during propagation for a
+time interval `t`
 """
-function propagate!(p::Particle{T}, t::T, offset::MArray{Tuple{4,4},T}) where {T<: AbstractFloat}
-    vx0=p.vel[1]
-    vy0=p.vel[2]
-    p.pos += [vx0*t, vy0*t]
+#linear
+function propagate_offset!(offset::MArray{Tuple{4,4},T}, t::T,
+                           p::Particle{T}) where T
     for k in 1:4
-        x = offset[:,k]
-        temp = [x[1] + t*x[3], x[2] + t*x[4], x[3], x[4]]
+        Γ = offset[:,k]
+        temp = [Γ[1] + t*Γ[3], Γ[2] + t*Γ[4], Γ[3], Γ[4]]
         offset[:,k] = temp
     end
 end
 
+#magnetic
+function propagate_offset!(offset::MArray{Tuple{4,4},T}, t::T,
+                           p::MagneticParticle{T}) where T
+    ω = p.omega
+    sω = sin(ω*t)
+    cω = cos(ω*t)
+    J = [1.0 0.0     sω/ω  (cω-1)/ω;
+         0.0 1.0  (1-cω)/ω     sω/ω;
+         0.0 0.0     cω         -sω;
+         0.0 0.0     sω          cω]
+    for k in 1:4
+        Γ = offset[:,k]
+        offset[:,k] = J*Γ
+    end
+end
+
 """
-    relocate(p::Particle, o::Obstacle, t, offset::MArray) -> newt
+    propagate!(p::AbstractParticle{T}, t::T, offset::MArray{Tuple{4,4},T})
```

Comment:
I think we should remove the documentation strings, as they are added to the documentation strings of the `propagate!` function.

This brings unecessary information to someone reading the docstrings for their own implementations. 

Can you simply comment the docstrings? Do not remove them, but make them comments instead.

Reply:
Should I do the same for the `resolvecollision` and `specular!` docstrings?

Reply:
Yes please, sorry I forgot.